### PR TITLE
SONARSLANG-50 Rule: Class names should comply with a naming convention

### PR DESCRIPTION
**Pull request by christophe-zurn-sonarsource**

### CODE REVIEW

File: slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangConverterTest.java [hunk 1]
```diff
@@ -143,14 +142,16 @@ public void variable_declaration_with_initializer() {
     assertTree(varDeclX).isNotEquivalentTo(converter.parse("myint var x = 0;").children().get(0));
     assertTree(varDeclX).isNotEquivalentTo(converter.parse("int var x = 1;").children().get(0));
     assertTree(varDeclX).isNotEquivalentTo(converter.parse("var x = 0;").children().get(0));
-    assertTree(varDeclX).isNotEquivalentTo(converter.parse("var x;").children().get(0));;
+    assertTree(varDeclX).isNotEquivalentTo(converter.parse("var x;").children().get(0));
+    ;
```
Comment (pynicolas) [inline slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangConverterTest.java]: Why?

File: sonar-slang-plugin/src/test/java/com/sonarsource/slang/kotlin/KotlinConverterTest.java [hunk 2]
```diff
@@ -167,11 +168,14 @@ public void testClassWithBody() {
...
   @Test
   public void testClassWithoutBody() {
-    Tree classA = kotlin("class A {}");
-    assertTree(classA).isInstanceOf(ClassDeclarationTree.class);
-    assertTree(classA).isEquivalentTo(kotlin("class A {}"));
-    assertTree(classA).isNotEquivalentTo(kotlin("class A constructor(){}"));
-    assertTree(classA).isNotEquivalentTo(kotlin("class B {}"));
+    Tree tree = kotlin("class A {}");
+    assertTree(tree).isInstanceOf(ClassDeclarationTree.class);
+    ClassDeclarationTree classA2 = (ClassDeclarationTree) tree;
```
Comment (pynicolas) [inline sonar-slang-plugin/src/test/java/com/sonarsource/slang/kotlin/KotlinConverterTest.java]: Why naming it `classA2`?

File: sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java [hunk 3]
```diff
@@ -214,12 +218,27 @@ private Tree convertElementToSlangAST(PsiElement element, TreeMetaData metaData)
...
+    IdentifierTree identifier = null;
+    String name = null;
+
+    if (nameIdentifier != null) {
+      name = nameIdentifier.getText();
+      identifier = createIdentifierTree(getTreeMetaData(nameIdentifier), name);
+    }
+    Boolean isInsideEnum = enumClassDeclaration.peekLast();
+    enumClassDeclaration.addLast(ktClass.isEnum());
+    Tree nativeClassDecl = createNativeTree(metaData, new KotlinNativeKind(ktClass, name), ktClass);
```
Comment (pynicolas) [inline sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java]: Why do we need to have the name in the native kind?

Comment (christophe-zurn-sonarsource) [reply sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java]: Good point. I tried removing it, but then equivalence test were failing. It seems for some reasons the `KtNameIdentifier` is not in the children of the `KtClass` object...
I fixed our converter to solve that problem in 8cddace7a47c66a5dde9c1cf14cf3da4e1d5e0e6

**Pull request merged**